### ISSUE TITLE
Window Capture not properly reporting current open windows.

### ISSUE THREAD

**Issue by jblevins1991** [obsproject/obs-studio#2160]
I noticed something when setting up a `programming stream profile`. The `Window Capture` Window dropdown never correctly reports the windows that are open. In one scenario Chrome and Visual Studio Code were both not being shown as options for Window Capture.

![Missing Chrome](https://user-images.githubusercontent.com/1040084/68093536-8f8bcf80-fe64-11e9-97af-6ba8fff39bb5.png)
![Missing VSCode](https://user-images.githubusercontent.com/1040084/68093537-91ee2980-fe64-11e9-90a1-edba2995b5e2.png)

**Comment by WizardCM** [comment]
There are two reasons for this:

1. The list is populated when the Properties window is opened, and doesn't refresh automatically
2. The list doesn't include windows that are minimised at time of populating

**Comment by jblevins1991** [comment]
> 
> 
> There are two reasons for this:
> 
>     1. The list is populated when the Properties window is opened, and doesn't refresh automatically
> 
>     2. The list doesn't include windows that are minimised at time of populating

I see. I do not think this is very intuitive. What file is this behavior written in? This codebase is very difficult to navigate.

**Comment by jblevins1991** [comment]
Can we change this functionality to where it checks for a .exe and if it doesn't exist it just doesn't record that source?

**Comment by Fenrirthviti** [comment]
I'm not sure I follow what you're asking. If the window doesn't exist, capture will fail and the source will be empty. We are using OS-provided APIs for enumerating available windows for capture, so there's not much else we can really do to control what shows up. We are using a custom blocklist to hide processes that will never be able to be captured (such as the start menu, lock screen, etc.), but the enumeration is handles by the APIs.

**Comment by jblevins1991** [comment]
What API is being used?

**Comment by ifohancroft** [comment]
> What API is being used?

I am not sure exactly how things are done on Windows, specially at the lower level, but generally, I suspect it's Windows' window manager that lets OBS know the list of windows. So I believe this should be reported to Microsoft and not OBS.

**Comment by bocekm** [comment]
I have exactly the same problem on Fedora 31 with Wayland. Only a handful of opened app show up in the dropdown of "Windows Capture (Xcomposite)". Even windows that were opened before starting OBS and that were maximized do not show up.

**Comment by jp9000** [comment]
Anyone able to compile and check current master branch?

**Comment by ifohancroft** [comment]
> Anyone able to compile and check current master branch?

Just did (on Linux though) and all windows get reported/captured properly:
A browser (Firefox)
Two terminals (Termite)
PDF Viewer (qpdfview)
FTP Client (Filezilla)
File manager (Thunar)
OBS itself
Text Editor (Geany)

Even some that probably shouldn't be captured/shown (the two xfce panels and the desktop itself)

My window manager is XFCE4's XFWM and I am running X as opposed to Wayland.

**Comment by kkartaltepe** [comment]
This issue is about windows specifically. Please use another issue for different platforms.

**Comment by wangshaohui1314** [comment]
> Anyone able to compile and check current master branch?

Current master branch works normally. Gennerally chrome and VS can be shown in list. 
It seems that he forgot to open window from minimized state.

**Comment by Fenrirthviti** [comment]
I'm closing this issue for now, as @wangshaohui1314 stated it sounds like this is just a misunderstanding on how the available windows are enumerated. Everything is tested to be working as expected. I would suggest proposing an RFC if you feel this should be changed, as this is not really common issue or problem we see in the wild.
